zhmcclient/_adapter.py

```diff
@@ -679,3 +679,58 @@ def list_assigned_partitions(self, full_properties=False, filter_args=None):
                     resource_obj.pull_full_properties()
 
         return resource_obj_list
+
+    @logged_api_call
+    def list_sibling_adapters(self, full_properties=False):
+        """
+        List the other Adapters on the same adapter card as this Adapter.
+
+        Some adapter cards are represented as multiple Adapter objects
+        (for example, 2-port FICON Express cards, or 2-port CNA cards).
+        This method lists the other Adapter objects that are on the same
+        adapter card as this Adapter object.
+
+        This is useful for example to determine the affected Adapter objects
+        when replacing the adapter card, or when changing the type of a FICON
+        Express adepter (see :meth:`~zhmcclient.Adapter.change_adapter_type`).
+
+        Authorization requirements:
+
+        * Object-access permission to this CPC.
+        * Object-access permission to any Adapter to be included in the result.
+
+        Parameters:
+
+          full_properties (bool):
+            Controls whether the full set of resource properties should be
+            retrieved, vs. only the short set as returned by the list
+            operation.
+
+        Returns:
+
+          : A list of :class:`~zhmcclient.Adapter` objects.
+
+        Raises:
+
+          :exc:`~zhmcclient.HTTPError`
+          :exc:`~zhmcclient.ParseError`
+          :exc:`~zhmcclient.AuthError`
+          :exc:`~zhmcclient.ConnectionError`
+        """
+        # This algorithm is based on the fact that physical adapter cards
+        # have their PCHIDs always within a range of 4 adjacent PCHIDs that
+        # start at a multiple of 4.
+
+        self_pchid = int(self.prop('adapter-id'), 16)
+        if self_pchid >= int('700', 16):
+            # Hipersocket adapter
```

Comment:
just on your comment: could be also an ISM or ICP adapter (they also have VCHIDs >700).

Reply:
With this code, I want to detect adapters without siblings, where sibling is defined as the set of other PCHIDs on the same same physical card. So I need to understand more details about ISM and ICP.

Reply:
Changed that to "Virtual adapter" and added more explanatory comments.